site/content/docs/multistep-checks/websockets.md

```diff
@@ -0,0 +1,160 @@
+---
+title: WebSockets
+weight: 20
+menu:
+  resources:
+    parent: "Multistep API checks"
+    identifier: "multistep-checks-websockets"
+cli: true
+---
+
+WebSockets provide a way to establish a persistent, full-duplex communication channel that operates over a single, long-held request, enabling real-time data transfer between a client and a server.
+
+To create a WebSocket check in Checkly, you'll need to write a script that establishes a WebSocket connection, sends messages, and validates responses.
+
+## Set Up the WebSocket Connection
+First, we'll establish a connection to the WebSocket server. Replace `wss://your.websocketserver.com` with the URL of your WebSocket server.
+
+{{< tabs "Setup WebSocket Connection" >}}
+{{< tab "Typescript" >}}
+```ts
+import { WebSocket } from 'ws'
+
+const url = 'wss://your.websocketserver.com';
+const ws = new WebSocket(url);
+```
+{{< /tab >}}
+{{< tab "Javascript" >}}
+```js
+const WebSocket = require('ws');
+
+const url = 'wss://your.websocketserver.com';
+const ws = new WebSocket(url);
+```
+{{< /tab >}}
+{{< /tabs >}}
+
+## Handle WebSocket Events
+Handle the key events like opening the connection, receiving messages, and encountering errors.
+
+{{< tabs "Handle WebSocket Events" >}}
+{{< tab "Javascript" >}}
+```js
+ws.on('open', () => {
+  console.log('WebSocket connection established.');
+  // Send initial message or subscription here, if necessary.
+});
+
+ws.on('message', (data) => {
+  console.log('Received message:', data);
+  // Add your message handling logic here.
+});
+
+ws.on('error', (error) => {
+  console.error('WebSocket error:', error);
+});
+```
+{{< /tab >}}
+{{< /tabs >}}
+
+## Sending and Receiving Messages
+Implement the logic for sending messages and processing the received responses.
+
+{{< tabs "Send/Receive WebSocket Messages" >}}
+{{< tab "Javascript" >}}
+```js
+function sendMessage(message) {
+  ws.send(message);
+}
+
+// Add logic to process received messages
+```
+{{< /tab >}}
+{{< /tabs >}}
+
+
+## Close the Connection
+Ensure to properly close the WebSocket connection after your checks.
+
+{{< tabs "Close WebSocket Connection" >}}
+{{< tab "Javascript" >}}
+```js
+ws.on('close', () => {
+  console.log('WebSocket connection closed.');
+});
+```
+{{< /tab >}}
+{{< /tabs >}}
+
+## Example: Monitoring a WebSocket Connection
+Here's the complete example script that demonstrates a WebSocket check. It sends a subscription message, prints out the echo'd response, handles different types of incoming messages and closes the connection after 5 seconds.
+
+{{< tabs "Full WebSocket Script" >}}
```

Comment:
@checklyalex question: wouldn't we want some assertions in the check? Or would you say they are superfluous?